tests/Temporalio.Tests/Client/TemporalClientScheduleTests.cs

```diff
@@ -0,0 +1,363 @@
+        };
+        await AssertMore.EqualEventuallyAsync(expectedIDs, async () =>
+        {
+            var actualIDs = new List<string>();
+            await foreach (var sched in Client.ListSchedulesAsync())
+            {
+                actualIDs.Add(sched.ID);
+            }
+            actualIDs.Sort();
+            return actualIDs;
+        });
+
+        // Delete when done
+        await DeleteAllSchedulesAsync();
+    }
+
+    [Fact]
+    public async Task CreateScheduleAsync_CalendarSpecDefaults_AreProper()
+    {
+        await AssertNoSchedulesAsync();
+
+        var handle = await Client.CreateScheduleAsync(
+            $"schedule-{Guid.NewGuid()}",
+            new(
+                Action: ScheduleActionStartWorkflow.Create(
+                    IKitchenSinkWorkflow.Ref.RunAsync,
+                    new KSWorkflowParams(new KSAction(Result: new("some result"))),
+                    new(id: $"workflow-{Guid.NewGuid()}", taskQueue: Env.KitchenSinkWorkerTaskQueue)),
+                Spec: new() { Calendars = new List<ScheduleCalendarSpec> { new() } })
+            {
+                State = new() { Paused = true },
+            });
+        var desc = await handle.DescribeAsync();
+        AssertMore.EqualAsJson(new ScheduleCalendarSpec(), desc.Schedule.Spec.Calendars.Single());
+        // Make sure that every next time has all zero time portion and is one day after the
+        // previous
+        Assert.Equal(10, desc.Info.NextActionTimes.Count);
+        for (var i = 0; i < desc.Info.NextActionTimes.Count; i++)
+        {
+            var nextTime = desc.Info.NextActionTimes.ElementAt(i);
+            Assert.Equal(0, nextTime.Second);
+            Assert.Equal(0, nextTime.Minute);
+            Assert.Equal(0, nextTime.Hour);
+            if (i > 0)
+            {
+                Assert.Equal(
+                    desc.Info.NextActionTimes.ElementAt(i - 1) + TimeSpan.FromDays(1),
+                    nextTime);
+            }
+        }
+
+        // Delete when done
+        await DeleteAllSchedulesAsync();
+    }
+
+    [Fact]
+    public async Task CreateScheduleAsync_TriggerImmediately_Succeeds()
+    {
+        await AssertNoSchedulesAsync();
+
+        // Create paused schedule that triggers immediately
+        var handle = await Client.CreateScheduleAsync(
+            $"schedule-{Guid.NewGuid()}",
+            new(
+                Action: ScheduleActionStartWorkflow.Create(
+                    IKitchenSinkWorkflow.Ref.RunAsync,
+                    new KSWorkflowParams(new KSAction(Result: new("some result"))),
+                    new(id: $"workflow-{Guid.NewGuid()}", taskQueue: Env.KitchenSinkWorkerTaskQueue)),
+                Spec: new())
+            {
+                State = new() { Paused = true },
+            },
+            new() { TriggerImmediately = true });
+
+        // Confirm workflow result
+        var desc = await handle.DescribeAsync();
```

Comment:
isn't this racey because the create call finished, but the schedule may not have started the action?